### PR TITLE
PBI 32-41: Filter Vendor Product by Name

### PR DESCRIPTION
**Pull request by edutjie**
## Describe your changes
- Implement get vendor product by name accessor
- Implement get vendor product by name service
- Implement order by for get vendor product accessor
- Implement order by for get vendor product service
- Query parameter wiring for both filter and order by
- UT for filter vendor product by name

## Issue ticket number and link
- ADU-32: https://linear.app/adudu/issue/ADU-32/filter-vendor-product-by-name-api
- ADU-41: https://linear.app/adudu/issue/ADU-41/ut-for-filter-vendor-product-by-name-api

### CODE REVIEW

File: internal/product/service.go [hunk 1]
```diff
@@ -7,15 +7,15 @@ import (
...
+	GetProductsByVendor(ctx context.Context, spec GetProductsByVendorSpec) ([]Product, error)
 }
 
 type ProductService struct {
 	productDBAccessor
 }
 
-func (p *ProductService) GetProductsByVendor(ctx context.Context, vendorID string) ([]Product, error) {
-	return p.productDBAccessor.GetProductsByVendor(ctx, vendorID)
+func (p *ProductService) GetProductsByVendor(ctx context.Context, spec GetProductsByVendorSpec) ([]Product, error) {
```
Comment (valeelim) [inline internal/product/service.go:17]:
```suggestion
func (p *ProductService) GetProductsByVendor(ctx context.Context, vendorID, spec GetProductsByVendorSpec) ([]Product, error) {
```

I think we should just keep the `vendorID` field. It enforces the need for `vendorID` for every client verbosely

File: internal/product/accessor.go [hunk 2]
```diff
@@ -31,8 +32,40 @@ type postgresProductAccessor struct {
...
 
-func (p *postgresProductAccessor) GetProductsByVendor(_ context.Context, vendorID string) ([]Product, error) {
-	rows, err := p.db.Query(getProductsByVendorQuery, vendorID)
+func (p *postgresProductAccessor) GetProductsByVendor(_ context.Context, spec GetProductsByVendorSpec) ([]Product, error) {
+	// Initialize clauses and arguments
+	var (
+		whereClauses []string
+		extraClauses []string
+		args         = []interface{}{spec.VendorID}
+		i            = 2
```
Comment (valeelim) [inline internal/product/accessor.go:41]: i think we should rename this to something more descriptive

File: internal/product/product.go [hunk 3]
```diff
@@ -23,3 +23,9 @@ type ProductVendor struct {
 	Product
 	vendors.Vendor
 }
+
+type GetProductsByVendorSpec struct {
+	VendorID string `json:"vendor_id"`
+	Name     string `json:"name"`
+	OrderBy  string `json:"order_by"`
```
Comment (valeelim) [inline internal/product/product.go:30]: any specific reason not to include `Order` here?

Comment (edutjie) [reply internal/product/product.go:30]: Because `Order` is in paginationSpec

**Review by valeelim — APPROVED** [review]
Looks Ganteng To Me

**Pull request merged**